### ISSUE TITLE
Definition of min/max causes conflict with inclusion of standard lib.

### ISSUE THREAD

**Issue by evoskuil** [zeromq/czmq#896]
From czmq++ build:
```
In file included from /home/eric/prefix/include/czmq.h:19:0,
                 from ./include/czmq++/message.hpp:25,
                 from src/message.cpp:20:
/usr/include/c++/4.8/bits/basic_string.h: In member function 'int std::basic_string<_CharT, _Traits, _Alloc>::compare(const std::basic_string<_CharT, _Traits, _Alloc>&) const':
/home/eric/prefix/include/czmq_prelude.h:396:25: error: expected unqualified-id before '(' token
 #   define min(a,b)     (((a)<(b))? (a): (b))
```

**Comment by hintjens** [comment]
OK, I'll remove these two macros. Seems to be a well-known issue with C++
and windows.h.

**Comment by jemc** [comment]
Resolved in PR #897

**Comment by evoskuil** [comment]
Thanks, yes I run into this on most of my VS ports.
